language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/definition/DefinitionUtil.java

```diff
@@ -63,25 +75,80 @@
         io.ballerina.tools.diagnostics.Location symbolLocation = symbol.getLocation().get();
         LinePosition startLine = symbolLocation.lineRange().startLine();
         LinePosition endLine = symbolLocation.lineRange().endLine();
         Position start = new Position(startLine.line(), startLine.offset());
         Position end = new Position(endLine.line(), endLine.offset());
         Range range = new Range(start, end);
 
-        if (symbol.getModule().isEmpty()) {
-            return Optional.empty();
+        return Optional.of(new Location(uri, range));
+    }
+
+    private static Optional<Path> getFilePathForDependency(String orgName, String moduleName,
+                                                           Project project, Symbol symbol) {
+        Collection<ResolvedPackageDependency> dependencies =
+                project.currentPackage().getResolution().dependencyGraph().getNodes();
+        Path filepath = null;
+        String sourceFile = symbol.getLocation().get().lineRange().filePath();
+        for (ResolvedPackageDependency depNode : dependencies) {
+            Package depPackage = depNode.packageInstance();
+            for (ModuleId moduleId : depPackage.moduleIds()) {
+                if (depPackage.packageOrg().value().equals(orgName) &&
+                        depPackage.module(moduleId).moduleName().toString().equals(moduleName)) {
+                    Module module = depPackage.module(moduleId);
+                    for (DocumentId docId : module.documentIds()) {
+                        if (module.document(docId).name().equals(sourceFile)) {
+                            filepath =
+                                    module.project().documentPath(docId).orElseThrow();
+                            break;
+                        }
+                    }
+                }
+            }
         }
-        Optional<String> uri = CommonUtil.getSymbolUriInProject(context, symbol);
-        
-        if (uri.isEmpty()) {
-            return Optional.empty();
+
+        return Optional.ofNullable(filepath);
+    }
+
+    private static Optional<Path> getFilePathForLanglib(String orgName, String moduleName,
+                                                        Project project, Symbol symbol) {
+        Package langLibPackage = project.projectEnvironmentContext().environment().getService(PackageCache.class)
+                .getPackages(PackageOrg.from(orgName), PackageName.from(moduleName)).get(0);
+        String sourceFile = symbol.getLocation().get().lineRange().filePath();
+
+        Path filepath = null;
+        for (ModuleId moduleId : langLibPackage.moduleIds()) {
+            Module module = langLibPackage.module(moduleId);
+            for (DocumentId docId : module.documentIds()) {
+                if (module.document(docId).name().equals(sourceFile)) {
+                    filepath =
+                            module.project().documentPath(docId).orElseThrow();
```

Comment:
Same as above. Can use `flatMap` instead of throwing.

Reply:
Used an alternative approach instead of using the flatmap